### PR TITLE
117 textinput component cant handle waveform pv input

### LINKED ISSUE #117: TextInput component can't handle waveform PV input
## binaryWaveform from example (Style Guide/Compound Components/ArrayContainer)
I have a TextInput component that receives a waveform PV like in the documentation:
```js
<TextInput
  pv="$(device):binaryWaveform"
  macros={{ "$(device)": "testIOC" }}
  usePvLabel={true}
  debug={true}
/>
```
image
When I try to change the value (lets say the last 0 to a 1), this happens:
image
Nothing else seems to break, just the value is broken.

## waveform with characters
It is even worse with waveform PVs that hold characters.
The goal is to have TextInput component that receives a waveform PV with character array:
```js
<TextInput
  pv="$(DEVICE):RawTiff:FilePath"
  macros={{ "$(DEVICE)": props.device }}
  label={"Save path"}
  useStringValue={true}
  debug={true}
/>
```
image
When I try to to change the value (lets say to /tempa), this happens:
image
And I get this error in the console:
`Warning: A component is changing a controlled input to be uncontrolled. This is likely caused by the value changing from a defined to undefined, which should not happen. Decide between using a controlled or uncontrolled input element for the lifetime of the component. More info: https://reactjs.org/link/controlled-components`
After doing some debugging, I found out is that `pv` variable in EpicsPV.js(React-Automation-Studio/ReactApp/src/components/SystemComponents/EpicsPV.js) gets value set to `undefined` when you try to update it.
The TextOutput component holding the same PV as TextInput also breaks the same way, when you try to update the value with TextInput.
Displaying the PV works fine, but something goes wrong when updating it.
The waveform with characters is fine once you refresh the page and displays the updated value, but breaks again as soon as you try to change it.

### PR DESCRIPTION
**Pull request by wduckitt**
close #117 

This updates the documentation to prevent the unintentional use case, I also changed the default prop to inculed a spacing of 1, finally I added another prop to show the array indices by default

### CODE REVIEW

**Comment by wduckitt** [comment]
I have added a patch to the pvServer, we were previously striping of the char_value of the waveform, as the  previous interpretation was that one would only want to display  an array of numerial values in a graph. As pointed out in #117 and #119 there is a use case to display and edit a waveform of char as a string with a TextInput or TextOuput. 

The impact on network traffic for the case of array of numerical values is minimal as pyEpics returns only the type of array as the char_value:
<array size=1000, type=time_double>
whilst value is an array of 1000 doubles.

For waveform of char, PyEpics returns the char_value as a tring.

Note: To correctly edit a waveform of bytes the TextInput or TextOuput must use the prop useStringValue={true}.

closes #119

File: ReactApp/src/components/CompoundComponents/ArrayContainer.js [hunk 1]
```diff
@@ -240,13 +245,18 @@ ArrayContainer.propTypes = {
    * or a subset of them.
    */
   registersLabel: PropTypes.arrayOf(PropTypes.string),
+   /**
+   * Directive to use the array indices as the labels,unless the registersLabel prop is defined
```
Comment (hanak) [inline ReactApp/src/components/CompoundComponents/ArrayContainer.js:249]: Space after comma: `...labels,unless...` --> `...labels, unless...`

File: ReactApp/src/components/CompoundComponents/ArrayContainer.js [hunk 2]
```diff
@@ -272,7 +282,8 @@ ArrayContainer.defaultProps = {
   direction: "vertical",
   labelPlacement: "top",
   itemMinWidth: 2,
-  spacing: 0,
+  spacing: 1,
+  showIndices:true
```
Comment (hanak) [inline ReactApp/src/components/CompoundComponents/ArrayContainer.js:286]: Space after ":" is missing compared to lines above.

**Review by hanak — APPROVED** [review]
Change looks good. I have added a few minor comments about formatting of comments. Would it be possible to take look at those prior merging the pull request, please?

**Comment by wduckitt** [comment]
closes #117 
closes #119

**Pull request merged**